libvirt/tests/src/storage/virsh_pool.py

```
@@ -1,27 +1,119 @@
+    current_release = releases[0]
+    image_path = image_path % current_release
+    params['image_path'] = image_path
+    _, _, _, sec_uuid = disk_base.DiskBase.create_rbd_disk_path(params)
+    test.log.debug("Step: rbd image is created.")
+    pool_dict = eval(params.get('pool_dict') % sec_uuid)
+    rbd_pool_obj = pool_xml.PoolXML()
+    rbd_pool_obj.setup_attrs(**pool_dict)
+    virsh.pool_define(rbd_pool_obj.xml, debug=True, ignore_status=False)
+    test.log.debug("Step: libvirt pool is created")
+
+
+def clean_rbd_pool(params, test):
+    """
+    Clean up image in rbd pool on ceph server and related resources
+
+    :param params: dict, test parameters
+    :param test: test object
+    """
+    disk_base.DiskBase.cleanup_rbd_disk_path(params)
+    test.log.debug("Step: rbd image is removed.")
+    virsh.pool_destroy(params.get('pool_name'), debug=True, ignore_status=True)
+    virsh.pool_undefine(params.get('pool_name'), debug=True, ignore_status=False)
+    test.log.debug("Step: libvirt pool is undefined")
+
+
+def verify_coredump():
+    """
+    Verify if a coredump file is produced which is unexpected
+    """
+    res = process.run("coredumpctl list",
+                      shell=True,
+                      verbose=True,
+                      ignore_status=True)
+    utlv.check_result(res,
+                      expected_fails='No coredumps found',
+                      check_both_on_error=True)
+
+
+def skip_case(params, test):
+    """
+    Skip the case if matched
+
+    :param params: dict, test parameters
+    :param test: test object
+    """
+    pool_type = params.get("pool_type")
+    if pool_type == 'rbd':
+        required_commands = eval(params.get('required_commands', '[]'))
+        if required_commands:
+            for cmd in required_commands:
+                if not shutil.which(cmd):
+                    test.cancel("This case requires the command '%s'" % cmd)
```

raise error instead of skip case

---

Update, thanks.